book/listings/todo_app/1/main.rs

```diff
@@ -0,0 +1,111 @@
+mod todo_object;
+mod todo_row;
+
+use glib::BindingFlags;
+use gtk::prelude::*;
+use gtk::Application;
+use gtk::ApplicationWindow;
+use gtk::{gio, glib};
+use gtk::{
+    CheckButton, ConstantExpression, Label, ListView, PolicyType, PropertyExpression,
+    ScrolledWindow, SignalListItemFactory, SingleSelection,
+};
+
+use todo_object::TodoObject;
+use todo_row::TodoRow;
+
+fn main() {
+    // Create a new application
+    let app = Application::new(Some("org.gtk.example"), Default::default());
+    app.connect_activate(build_ui);
+
+    // Run the application
+    app.run();
+}
+
+fn build_ui(application: &Application) {
+    // Create a window
+    let window = ApplicationWindow::builder()
+        .application(application)
+        .title("My GTK App")
+        .default_width(600)
+        .default_height(300)
+        .build();
+
+    let model = gio::ListStore::new(TodoObject::static_type());
+    for i in 0..1000 {
+        model.append(&TodoObject::new(i.to_string(), true));
+    }
+
+    let factory = SignalListItemFactory::new();
+    factory.connect_setup(move |_, list_item| {
+        // Create todo row
+        let todo_row = TodoRow::new();
+        list_item.set_child(Some(&todo_row));
+    });
+
+    let bindings_map = std::rc::Rc::new(std::cell::RefCell::new(std::collections::HashMap::new()));
```

Comment:
@bilelmoussaoui Using a hashmap to store the bindings is the only working thing I came up so far